### ISSUE TITLE
Selecting artboard after dismissing Export dialog causes error

### ISSUE THREAD

**Issue by ktaki** [adobe-photoshop/spaces-design#2275]
Create an artboard. Add a shape. Add export setting to the artboard. Open Export dialog. Do export. Dismiss dialog. Click the artboard name to select the artboard. This often (not always) fails with the following error. (This is a fatal error. DS needs to be reset.)

![screen shot 2015-08-28 at 5 23 28 pm](https://cloud.githubusercontent.com/assets/2029597/9559546/f343c46c-4da9-11e5-9b99-448a66e6a7da.png)

**Comment by iwehrman** [comment]
I'm hitting something similar with the following steps:

1. Create a new non-AB document. Unlock the background layer. Configuring an export for that layer.
2. Repeat.
3. Select first document.

Result:
```
Error: Failed to register dialog: datalist-layerExportAsset-scale2-0 already exists.
    at Fluxxor.createStore.registerDialog (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/stores/dialog.js:131:23)
    at React.createClass.componentWillMount (file://./js/jsx/shared/Dialog.jsx:109:44)
    at chainedFunction [as componentWillMount] (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:6044:9)
    at ReactCompositeComponentMixin.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:6884:14)
    at ReactDefaultPerf.measure (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:10068:19)
    at ReactPerf.measure.wrapper [as mountComponent] (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:13968:31)
    at Object.ReactReconciler.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:14781:35)
    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:13414:44)
    at ReactDOMComponent.Mixin._createContentMarkup (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:8207:32)
    at ReactDOMComponent.Mixin.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:8117:12)
    at Object.ReactReconciler.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:14781:35)
    at ReactCompositeComponentMixin.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:6903:34)
    at ReactDefaultPerf.measure (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:10047:23)
    at ReactPerf.measure.wrapper [as mountComponent] (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:13968:31)
    at Object.ReactReconciler.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:14781:35)
    at ReactCompositeComponentMixin.mountComponent (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:6903:34)
From previous event:
    at Object.FluxController._makeActionReceiver.Object.create.dispatchAsync.value (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:400:44)
    at Object.updateLayerExportAsset (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/actions/export.js:238:21)
    at FluxController._applyAction (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:453:36)
    at Object.FluxController._makeActionReceiver.Object.create.transfer.value (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:387:33)
    at Object.<anonymous> (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/actions/export.js:322:29)
From previous event:
    at Object.addLayerAsset (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/actions/export.js:321:14)
    at FluxController._applyAction (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:453:36)
    at null.<anonymous> (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:546:37)
From previous event:
    at null.<anonymous> (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:542:22)
    at AsyncDependencyQueue._processNext (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/util/async-dependency-queue.js:291:27)
    at AsyncDependencyQueue.push (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/util/async-dependency-queue.js:143:14)
    at Object.FluxController._synchronize (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:525:42)
    at React.createClass._addAssetClickHandler (file://./js/jsx/sections/export/ExportPanel.jsx:120:43)
    at Object.executeDispatch (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:3311:21)
    at SimpleEventPlugin.executeDispatch (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:17428:40)
    at forEachEventDispatch (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:3299:5)
    at Object.executeDispatchesInOrder (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:3320:3)
    at executeDispatchesAndRelease (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:2693:22)
    at Array.forEach (native)
    at forEachAccumulated (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:19430:9)
    at Object.EventPluginHub.processEventQueue (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:2900:5)
    at runEventQueueInBatch (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:11217:18)
    at Object.ReactEventEmitterMixin.handleTopLevel [as _handleTopLevel] (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:11243:5)
    at handleTopLevelImpl (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:11329:24)
    at ReactDefaultBatchingStrategyTransaction.Mixin.perform (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:18402:20)
    at Object.ReactDefaultBatchingStrategy.batchedUpdates (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:9669:19)
    at Object.batchedUpdates (file:///Users/ian/Desktop/photoshop-Release_x86_64/Plug-Ins/Spaces/www/bower_components/react/react-with-addons.js:16633:20)
```

**Comment by mcilroyc** [comment]
FYI this is fixed in a branch that is waiting on another PR to be merged.

**Comment by iwehrman** [comment]
:tada:

**Comment by ktaki** [comment]
Yes, this seems to be fixed. I still see the warning: Failed to deregister dialog...
Is this not a concern?

**Comment by mcilroyc** [comment]
The fix for this is still open (2319).  I *will* be concerned about "failed to deregister.." *after* that PR has merged.

**Comment by iwehrman** [comment]
FBNC to @ktaki now that #2319 is merged.

**Comment by ktaki** [comment]
Verified the fix.
